### PR TITLE
Add Ruby parametric tests

### PR DESCRIPTION
**Pull request by marcotc**
## Description

The first implementation of the Ruby gRPC server to support parametric system testing.

The tests where the Ruby tracer does not adhere to the system test requirements have been skipped, with a description left on what the underlying issue is.

Our team will follow up with fixes to the test had to be skipped.

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [x] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [x] CI is passing
- [x] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

**Review by ivoanjo — COMMENTED** [review]
Left a few notes, although it's hard for me to review `server.rb`, I'm missing a lot of context :sweat_smile:

File: parametric/test_headers_tracecontext.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ def test_traceparent_included_tracestate_missing(test_agent, test_library):
     "golang does not reconcile duplicate http headers, if duplicate headers received the propagator will not be used",
 )
 @pytest.mark.skip_library("java", "Issue: tracecontext is not merged yet")
+@pytest.mark.skip_library("ruby", "Ruby doesn't support case-insensitive distributed headers")
```
[the remark this answers is not in the document]
Comment (marcotc) [reply parametric/test_headers_tracecontext.py:69]: Yes.

File: parametric/apps/ruby/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+To debug the server locally, run:
+```sh
+bundle install
+./generate_proto.sh
+DEBUG=1 bundle exec ruby server.rb
+```
+
+You'll be presented with a Ruby REPL, with the gRPC server running in the same process.
+
+A `client` object will be available for you to make gRPC requests to the server.
```
[the remark this answers is not in the document]
Comment (marcotc) [reply parametric/apps/ruby/README.md:29]: Done!

File: parametric/conftest.py [hunk 3]
```diff
@@ -411,7 +444,7 @@ def docker_run(
     r = subprocess.run(_cmd, stdout=log_file, stderr=log_file)
     if r.returncode != 0:
         pytest.fail(
-            "Could not start docker container %r with image %r, see the log file %r" % (name, image, log_file),
+            "Could not start docker container %r with image %r, see the log file %r" % (name, image, log_file.read()),
```
Comment (Kyle-Verhoog) [inline parametric/conftest.py:447]:
```suggestion
            "Could not start docker container %r with image %r, see the log file %r" % (name, image, log_file),
```

I think this log message is meant to include the name of the log file, not its contents.

**Review by Kyle-Verhoog — COMMENTED** [review]
just one small thing, otherwise LGTM! thanks @marcotc for adding Ruby support! 👏

File: parametric/conftest.py (hunk 3 above)
Comment (marcotc) [reply parametric/conftest.py:447]: For me, it returns something opaque, but I'm ok reverting this.

**Review by cbeauchesne — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**